src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,404 @@
+    :func:`torch_xla.distributed.xla_fully_sharded_data_parallel.XlaFullyShardedDataParallel` method.
+
+    For more information check out https://github.com/pytorch/xla/blob/master/docs/fsdp.md
+    """
+
+    def __init__(
+        self,
+        accelerator: Optional[Accelerator] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision: Optional[Precision] = None,
+        state_dict_type: Literal["full", "sharded"] = "sharded",
+        sync_module_states: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            checkpoint_io=checkpoint_io,
+            precision=precision,
+        )
+        self._backward_sync_control = _XLAFSDPBackwardSyncControl()
+
+        self._fsdp_kwargs = kwargs
+        self._state_dict_type = state_dict_type
+        self._sync_module_states = sync_module_states
+
+    def setup_environment(self) -> None:
+        from torch_xla.experimental.pjrt import using_pjrt
+
+        assert self.parallel_devices is not None
+        if using_pjrt() and len(self.parallel_devices) == 1:
+            # spawning only 1 device with PjRT is not supported:
+            # https://github.com/Lightning-AI/lightning/pull/17408#discussion_r1170671732
+            raise NotImplementedError(
+                "The `XLAFSDPStrategy` does not support running on a single device with the PjRT runtime."
+                " Try using all devices or the `SingleTPUStrategy` strategy"
+            )
+
+        self._launched = True
+        rank_zero_only.rank = self.global_rank
+        super().setup_environment()
+
+    def setup_module_and_optimizers(
+        self, module: Module, optimizers: List[Optimizer]
+    ) -> Tuple[Module, List[Optimizer]]:
+        """Returns NotImplementedError since for XLA FSDP optimizer setup must happen after module setup."""
+        raise NotImplementedError(
+            f"The `{type(self).__name__}` does not support the joint setup of module and optimizer(s)."
+            " Please do it in this order: Create the model, call `setup_module`, create the optimizer,"
+            " call `setup_optimizer`."
+        )
+
+    def setup_module(self, module: Module) -> Module:
+        from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as XLAFSDP
+
+        if any(isinstance(mod, XLAFSDP) for mod in module.modules()):
+            if "auto_wrap_policy" in self._fsdp_kwargs:
+                rank_zero_warn(
+                    "A XLAFSDP `auto_wrap_policy` is set, but the model is already wrapped. The policy will be ignored."
+                )
+                del self._fsdp_kwargs["auto_wrap_policy"]
+        else:
+            if self._sync_module_states:
+                import torch_xla.core.xla_model as xm
+
+                xm.broadcast_master_param(module)
```

Comment:
The import path was moved in 2.1.

```suggestion
                if _XLA_GREATER_EQUAL_2_1:
                    from torch_xla.core.xla_model import broadcast_master_param
                else:
                    from torch_xla.experimental.pjrt import broadcast_master_param

                broadcast_master_param(module)
```